src/librustc_typeck/check/mod.rs

```diff
@@ -3057,20 +3057,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
             return field_ty;
         }
 
-        if tuple_like {
-            type_error_struct!(self.tcx().sess, expr.span, expr_t, E0612,
```

Comment:
I feel like this error is a bit different than the others. Not sure if this is a good idea to merge it as well...